### DISCUSSION TITLE
Help: UQL to extract data from API

### DISCUSSION THREAD

**Discussion by Nik0lax** [grafana/grafana-infinity-datasource#650] [Q&A]
Hi, I'm trying to use UQL to build a Time series chart to monitor real-time results. However, the JSON response from the API I'm using is returned like this:
{
"transactions": [
  {
    "id": "xxxx",
    "status": [
     {
        "createdAt": "2023-08-01T00:00:00-03:00",
        "type": "REQUESTED"
      },
      {
        "createdAt": "2023-08-01T00:00:05Z",
        "type": "APPROVED"
      },
      {
        "createdAt": "2023-05-16T00:00:10Z",
        "type": "CANCELED"
      } 
}
I would like to know how can I access the last object inside "statuses", like last "type", using UQL. I checked the documentation, but I didn't succeed. Thank you for all the help.

**Comment by paige-muon** [comment]
Converting your JSON to valid JSON:

```
{
  "transactions": [
    {
      "id": "xxxx",
      "status": [
        {
          "createdAt": "2023-08-01T00:00:00-03:00",
          "type": "REQUESTED"
        },
        {
          "createdAt": "2023-08-01T00:00:05Z",
          "type": "APPROVED"
        },
        {
          "createdAt": "2023-05-16T00:00:10Z",
          "type": "CANCELED"
        }
      ]
    }
  ]
}
```
The UQL I used was
```
parse-json
| scope "transactions"
| mv-expand "status"
| project "id", "created"="status.createdAt", "type"="status.type"
```
Which gave me a simple table with columns for id, created, and type:
![image](https://github.com/yesoreyeram/grafana-infinity-datasource/assets/115116191/0da389dd-e401-4661-ac72-89108ddbf552)

And it also scales if you have multiple transactions. I added another id entry to the example JSON and parsed with the same UQL and got this result:
```
{
  "transactions": [
    {
      "id": "xxxx",
      "status": [
        {
          "createdAt": "2023-08-01T00:00:00-03:00",
          "type": "REQUESTED"
        },
        {
          "createdAt": "2023-08-01T00:00:05Z",
          "type": "APPROVED"
        },
        {
          "createdAt": "2023-05-16T00:00:10Z",
          "type": "CANCELED"
        }
      ]
    },
    {
      "id": "yyyy",
      "status": [
        {
          "createdAt": "2023-09-01T00:00:00-03:00",
          "type": "REQUESTED"
        },
        {
          "createdAt": "2023-09-01T00:00:05Z",
          "type": "APPROVED"
        },
        {
          "createdAt": "2023-09-16T00:00:10Z",
          "type": "CANCELED"
        }
      ]
    }
  ]
}
```
![image](https://github.com/yesoreyeram/grafana-infinity-datasource/assets/115116191/7f5eac08-dbf2-4ce6-9da3-991c6b6c3af8)
